enumerable.rb

```diff
@@ -0,0 +1,171 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield to_a[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return self unless block_given?
+
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each { |x| arr << x if yield x }
+    arr
+  end
+
+  def my_all?(pattern = nil)
+    val = true
+    if pattern.nil?
+      block_given? ? my_each { |x| val = false unless yield x } : val = my_all? { |x| !x.nil? && (x != false) }
+    else
+      my_each { |x| val = false unless pattern === x }
+    end
+    val
+  end
+
+  def my_any?(pattern = nil)
+    val = false
+    if pattern.nil?
+      block_given? ? my_each { |x| val = true if yield x } : val = my_any? { |x| !x.nil? && (x != false) }
+    else
+      my_each { |x| val = true if pattern === x }
+    end
+    val
+  end
+
+  def my_none?(pattern = nil)
+    val = true
+    if pattern.nil?
+      block_given? ? my_each { |x| val = false if yield x } : val = my_any? { |x| x == true }
+    else
+      my_each { |x| val = false if pattern === x }
+    end
+    val
+  end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
  ```ruby
    array = [nil, false, nil, false]
    p array.my_none? == array.none? # false
  ```